frb_codegen/src/config.rs

```diff
@@ -114,53 +194,81 @@ pub fn parse(raw: RawOpts) -> Opts {
+    // dart root(s)
+    let dart_roots = match raw.dart_root {
+        Some(dart_roots) => dart_roots
+            .into_iter()
+            .map(|each_path| Some(canon_path(&each_path)))
+            .collect::<Vec<_>>(),
+        None => dart_output_paths
+            .iter()
+            .map(|each_dart_output_path| fallback_dart_root(&each_dart_output_path).ok())
+            .collect::<Vec<_>>(),
     };
 
-    Opts {
-        rust_input_path,
-        dart_output_path: canon_path(&raw.dart_output),
-        dart_decl_output_path: raw
-            .dart_decl_output
-            .as_ref()
-            .map(|s| canon_path(s.as_str())),
-        c_output_path,
-        rust_crate_dir,
-        rust_output_path,
-        class_name,
-        dart_format_line_length: raw.dart_format_line_length.unwrap_or(80),
-        skip_add_mod_to_lib: raw.skip_add_mod_to_lib,
-        llvm_path: raw.llvm_path.unwrap_or_else(|| {
-            vec![
-                "/opt/homebrew/opt/llvm".to_owned(), // Homebrew root
-                "/usr/local/opt/llvm".to_owned(),    // Homebrew x86-64 root
-                // Possible Linux LLVM roots
-                "/usr/lib/llvm-9".to_owned(),
-                "/usr/lib/llvm-10".to_owned(),
-                "/usr/lib/llvm-11".to_owned(),
-                "/usr/lib/llvm-12".to_owned(),
-                "/usr/lib/llvm-13".to_owned(),
-                "/usr/lib/llvm-14".to_owned(),
-                "/usr/lib/".to_owned(),
-                "/usr/lib64/".to_owned(),
-                "C:/Program Files/llvm".to_owned(), // Default on Windows
-                "C:/Program Files/LLVM".to_owned(),
-                "C:/msys64/mingw64".to_owned(), // https://packages.msys2.org/package/mingw-w64-x86_64-clang
-            ]
-        }),
-        llvm_compiler_opts: raw.llvm_compiler_opts.unwrap_or_else(|| "".to_string()),
-        manifest_path,
-        dart_root,
-        build_runner: !raw.no_build_runner,
-        exclude_sync_execution_mode_utility: raw
-            .exclude_sync_execution_mode_utility
-            .unwrap_or(false),
+    // exclude_sync_execution_mode_utility(s)
+    // (due to conflict of dart api of multiple blocks, keep it temporarily)
+    let exclude_sync_execution_mode_utilities = match raw.exclude_sync_execution_mode_utility {
+        Some(exclude_sync_execution_mode_utilities) => exclude_sync_execution_mode_utilities,
+        None => {
+            let mut exclude_sync_execution_mode_utilities = Vec::new();
+            for i in 0..rust_input_paths.len() {
+                if i == 0 {
+                    exclude_sync_execution_mode_utilities.push(false);
+                } else {
+                    exclude_sync_execution_mode_utilities.push(true);
+                }
+            }
+            exclude_sync_execution_mode_utilities
+        }
+    };
+
+    // build Opt for each rust api block
+    let mut opts = Vec::new();
+    for i in 0..rust_input_paths.len() {
```

Comment:
What about https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.zip_eq and iterators, instead of brute-force loop index i?

Reply:
I don't see it. Is there any benefit with `itertools` this way?

Reply:
Personally I like functional programming style instead of create-a-new-vec-and-push-things-to-it. But anyway this sounds ok as well.

Reply:
> Personally I like functional programming style instead of create-a-new-vec-and-push-things-to-it. But anyway this sounds ok as well.

I changed it into another functional style  in 55f2ef0aded946195114b076e329a631edab04b5.